deno_dist/validator/validator.ts

```diff
@@ -67,7 +67,7 @@ export const validator = <
 
     switch (target) {
       case 'json':
-        if (!contentType || !contentType.startsWith('application/json')) {
+        if (!contentType || !contentType.match(/^application\/[a-z-]*\+json(\s*;.*)?$/)) {
```

Comment:
* You should use the `test` instead of the `match` if you don't capture strings.
* `^application\/[a-z-]*\+` should be `application\/([a-z-]+\+)*`.
* The last `(\s*;.*)?` is necessary?

Reply:
Yeah... that last part was needed.

Some clients (like android's retrofit) sends the content type as: application/json; charset=UTF-8. Without that last bit the content-type header doesn't pass the regex.

Retrofit's discussion on this: https://github.com/square/retrofit/issues/794